src/transformers/models/poolformer/configuration_poolformer.py

```diff
@@ -0,0 +1,142 @@
+# coding=utf-8
+# Copyright Sea AI Labs and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PoolFormer model configuration """
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+POOLFORMER_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "seaailabs/poolformer_s12": "https://huggingface.co/seaailabs/poolformer_s12/resolve/main/config.json",
+    # See all PoolFormer models at https://huggingface.co/models?filter=poolformer
+}
+
+
+class PoolFormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a :class:`~transformers.PoolFormerModel`.
+    It is used to instantiate an PoolFormer model according to the specified arguments, defining the model
+    architecture. Instantiating a configuration with the defaults will yield a similar configuration to that of
+    the PoolFormer `seaailabs/poolformer_s12 <https://huggingface.co/seaailabs/poolformer_s12>`__ architecture.
+
+    Configuration objects inherit from  :class:`~transformers.PretrainedConfig` and can be used
+    to control the model outputs. Read the documentation from  :class:`~transformers.PretrainedConfig`
+    for more information.
+
+
+    Args:
+        vocab_size (:obj:`int`, `optional`, defaults to 30522):
+            Vocabulary size of the PoolFormer model. Defines the number of different tokens that can be represented by the
+            :obj:`inputs_ids` passed when calling :class:`~transformers.PoolFormerModel` or
+            :class:`~transformers.TFPoolFormerModel`.
```

Comment:
The docstrings here seem to be taken from the CookieCutter template. You can replace them by those of ViT (and update for PoolFormer).